Survey/Survey/Sources/Data/NetworkAPI/Core/RequestConfiguration.swift

```diff
@@ -0,0 +1,37 @@
+//
+//  RequestConfiguration.swift
+//  Survey
+//
+//  Created by Taher on 4/11/22.
+//
+
+import Foundation
+
+protocol RequestConfiguration {
```

Comment:
Optional: I used to apply Moya in my previous project, and I saw they have a good definition for request configuration using Enum. With this, you can implement API faster and look much prettier. What do you think?

**Definition:**
```
enum APIRequest: RequestConfiguration {
     case login, logout, register
}
```

**Usage**

extension APIRequest {

    var baseURL: String {
        "https://survey-api.nimblehq.co/" // TODO: Read from config file in integration task
    }

    var endpoint: String {
        switch self {
             case .login: return "api/v1/oauth/token" // TODO: Read from config file in integration task
             case .logout: return ""
             default: return .empty
        }        
    }

    var method: URLRequest.HttpMethod {
        switch self {
             case .login, .logout: return .post
             default: return .get
        }        
    }
}

Reply:
As for the first time I am using combine, so I wan to use the native URLSession for this time, thats why I didn't use Alamofire or Moya. Will use Moya in future surely.

Reply:
Yes, I got your idea, I didn't suggest you use Moya suggest you use Enum like Moya to define the Request, and this is optional 😁

Reply:
thanks .. got it now.. updated in this [commit](https://github.com/Shayokh144/ic-survey-swiftui-combine/pull/43/commits/1aab549f658f6030d71ed30af0d2429788c73d4b)